cmd/gotestmd/gotestmd.go

```diff
@@ -60,22 +73,73 @@ func New() *cobra.Command {
 			}
 
 			suites := g.Generate(linkedExamples...)
-			for _, suite := range suites {
-				dir, _ := filepath.Split(suite.Location)
-				_ = os.MkdirAll(dir, os.ModePerm)
-				err := os.WriteFile(suite.Location, []byte(suite.String()), os.ModePerm)
-				if err != nil {
-					return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
-				}
+
+			if !bash {
+				return processGoSuites(suites)
 			}
 
-			return nil
+			matchRegex, err := regexp.Compile(match)
+			if err != nil {
+				return err
+			}
+
+			return processBashSuites(suites, matchRegex)
 		},
 	}
 
+	gotestmdCmd.Flags().Bool("bash", false, "generates bash scripts for tests. Can be used only with --match flag")
+	gotestmdCmd.Flags().String("match", "", "regex for matching suite or test name. Can be used only with --bash flag")
+
 	return gotestmdCmd
 }
 
+func processGoSuites(suites []*generator.Suite) error {
+	for _, suite := range suites {
+		dir, _ := filepath.Split(suite.Location)
+		_ = os.MkdirAll(dir, os.ModePerm)
+		err := os.WriteFile(suite.Location, []byte(suite.String()), os.ModePerm)
+		if err != nil {
+			return errors.Errorf("cannot save suite %v, : %v", suite.Name(), err.Error())
+		}
+	}
+
+	return nil
+}
+
+func processBashSuites(suites []*generator.Suite, matchRegex *regexp.Regexp) error {
+	for _, suite := range suites {
+		if !matchRegex.MatchString(suite.Name()) {
```

Comment:
`suite.Name()` returns a short name.
For example, for all of `/interdomain/dns`, `/k8s_monolith/external_nsc/dns`, `/k8s_monolith/external_nse/dns`, `/multicluster/dns` it returns `dns`, and it's impossible to specify a certain suite that I want to use for generation.

We can match the regexp against the name of the generated file instead: `suite.Location`.
Though, if possible, it would be better to match against the source file location.